src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -484,6 +484,10 @@ int BaseCouplingScheme::getTimeWindows() const
 
 double BaseCouplingScheme::getNextTimeStepMaxSize() const
 {
+  if (!isCouplingOngoing()) {
+    return 0.0; // if coupling is not ongoing (i.e. coupling scheme reached end of window) the maximum time step size is zero
+  }
```

Comment:
Shouldn't this be the result of `_maxTime-time` anyway?

Reply:
`_maxTime` is not necessarily given. You may not provide a `_maxTime`, if you provide `_maxTimeWindows`. Then `_maxTime = UNDEFINED_MAX_TIME`.